lms/djangoapps/mobile_api/mobile_platform.py

```diff
@@ -58,6 +58,29 @@ class IOS(MobilePlatform):
                         r'\(Build [0-9a-zA-Z]*\)\)')
     NAME = "iOS"
 
+    @classmethod
+    def get_user_app_platform(cls, user_agent, user_agent_regex):
+        """
+        Replaces build number(3172) with app version(2.26.3) and
+        Returns platform instance if user_agent matches with USER_AGENT_REGEX
+
+        Arguments:
+            user_agent (str): user-agent for mobile app making the request.
+            user_agent_regex (regex str): Regex for user-agent valid for any type pf mobile platform.
+
+        Returns:
+           An instance of class passed (which would be one of the supported mobile platform
+           classes i.e. PLATFORM_CLASSES) if user_agent matches regex of that class else returns None
+        """
+
+        # Replace Build number 3172 with app version number 2.26.3
+        # so that we can enable upgrade banner and compare this version with other versions
+        # For details visit the ticket https://openedx.atlassian.net/browse/LEARNER-8639
+        sub_regex = r'(3172)(; OS Version [0-9.]+ \(Build [0-9a-zA-Z]*\)\))'
+        user_agent = re.sub(sub_regex, r'2.26.3\2', user_agent)
```

Comment:
What does `\2` do? (out of curiosity, I'm sure it's fine)

In `[0-9.]+`, do you not need a backslash for the period? Otherwise it'd match any character, yeah?

Reply:
\2 puts second matched group i.e. ; OS Version [0-9.]+ \(Build [0-9a-zA-Z]*\)\)) in the output string.

[According to docs](https://docs.python.org/3/library/re.html#:~:text=Special%20characters%20lose%20their%20special%20meaning%20inside%20sets.%20For%20example%2C%20%5B(%2B*)%5D%20will%20match%20any%20of%20the%20literal%20characters%20%27(%27%2C%20%27%2B%27%2C%20%27*%27%2C%20or%20%27)%27.): Special characters lose their special meaning inside sets. For example, [(+*)] will match any of the literal characters '(', '+', '*', or ')'.

Reply:
:+1: